### PR TITLE
Document hard link requirement

### PR DESCRIPTION
**Pull request by snh**
While we do mention hard links later on in the README, we don't currently make it clear that backup snapshots must be written to a filesystem which supports hard links.

/cc @github/backup-utils

### CODE REVIEW

**Comment by rubiojr** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -48,7 +48,8 @@ Backup host software requirements are modest: Linux or other modern Unix
 operating system with bash and [rsync][4] v2.6.4 or newer.
 
 The backup host must be able to establish network connections outbound to the
-GitHub appliance over SSH. TCP port 122 is used to backup GitHub Enterprise 2.0 or newer instances, and TCP port 22 is used for older versions (11.10.34X).
+GitHub appliance over SSH. TCP port 122 is used to backup GitHub Enterprise 2.0
```
Comment (michaeltwofish) [inline README.md]: Rogue newline 💄

File: README.md [hunk 2]
```diff
@@ -57,6 +58,9 @@ patterns of the GitHub appliance. We recommend allocating at least 5x the amount
 of storage allocated to the primary GitHub appliance for historical snapshots
 and growth over time.
 
+As Repository, Search, and Pages data is stored efficiently via hard links,
+backup snapshots must be written to a filesystem which supports hard links.
```
Comment (michaeltwofish) [inline README.md]: Mentioning the specific data that uses hard links makes this fragile to future changes, so I'd leave the specifics out. A link to a good, concise description of hard links would help too.

How about:

> The backup utilities use [hard links]() to store data efficiently, so the backup filesystem used must support hard links.

**Comment by michaeltwofish** [comment]
A couple of small comments, but I'm 👍 👍 on this change.

File: README.md (hunk 1 above)
Comment (snh) [inline README.md]: This was intentional, as the existing line was inconsistent with the formatting style of the document.

Comment (michaeltwofish) [inline README.md]: derp, h'okay.

**Comment by michaeltwofish** [comment]
🚢

**Pull request merged**